cogent/src/Cogent/Dargent/Surface.hs

```diff
@@ -53,13 +59,18 @@ data DataLayoutExpr' e
 #endif
   | Offset  (DataLayoutExpr' e) DataLayoutSize
   | RepRef  RepName
+  | LVar    DLVarName
   | Ptr
   deriving (Show, Data, Eq, Ord)
 
 -- We use 'tying the knot' here so we can make single level layouts later
 newtype DataLayoutExpr = DL { unDataLayoutExpr :: DataLayoutExpr' DataLayoutExpr }
   deriving (Show, Data, Eq, Ord)
 
+data TCDataLayout = TL { unTCDataLayout :: DataLayoutExpr' TCDataLayout }
+                  | TLU Int
+                  deriving (Show, Data, Eq, Ord)
```

Comment:
Any `TC*` types should be defined in `TypeCheck.Base`.

Reply:
well we need to export `DataLayoutExpr'` then...

Reply:
I don't think you can avoid it.

> well we need to export `DataLayoutExpr'` then

I don't think you can avoid it.